ecl/hthor/hthorkey.cpp

```diff
@@ -1454,6 +1454,8 @@ const void *CHThorIndexCountActivity::nextRow()
                             helper.onLimitExceeded();
                     }
 
+                    if (totalCount && (helper.getFlags() & TIRaggregateexists)) // EXISTS only needs to count one row
+                        break;
```

Comment:
can you confirm this actually makes a difference?
e.g. what happens without this with EXISTS, when does it break out (what is rowLimit/choosenLimit with EXISTS) ?